sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -4061,10 +4061,30 @@ object SessionWindowing extends Rule[LogicalPlan] {
           case s: SessionWindow => sessionAttr
         }
 
+        def isGapInvalid(child: Expression): Boolean = {
+          val calendarInterval = IntervalUtils
+            .safeStringToInterval(UTF8String.fromString(child.toString))
+          calendarInterval == null ||
+            calendarInterval.months + calendarInterval.days + calendarInterval.microseconds <= 0
+        }
+
+        val filterTimeSize = gapDuration match {
```

Comment:
Which cases this targets to? I only see the case of StringType (meaning static gap) which is going to be casted as CalendarIntervalType, but please correct me if I'm missing something.

Please make sure the code tightly picks up with the cases. For me, current pattern matching does not seem to be intuitive to understand. If you feel this code does not seem to be intuitive due to the fact you are disassembling Cast here, you can just deal with `session.gapDuration` instead, which is not yet wrapped.

Reply:
@HeartSaVioR
 Thanks for review!

I thought it is that there are three cases, including static gap duration, case when expression and user-defined function. When UDF, it needs to be filtered. Otherwise, there are gap duration and case when. In the case of case when, iterate its child nodes.I'm not quite sure if there are forms other than these three cases.